core/api/src/main/java/org/trellisldp/api/ImmutableDataService.java

```diff
@@ -28,11 +29,13 @@
  */
 public interface ImmutableDataService<T> extends RetrievalService<T> {
 
+
     /**
-     * @param identifier the identifier under which to persist a resource
+     * @param identifier the identifier under which to persist a dataset
      * @param session the session context for this operation
-     * @param resource a resource to persist
+     * @param dataset a dataset to persist
      * @return whether the resource was successfully persisted
      */
-    Future<Boolean> add(IRI identifier, Session session, T resource);
+    Future<Boolean> add(IRI identifier, Session session, Dataset dataset);
```

Comment:
Since the interface is no longer using the generic `<T>` type, could we remove that from the definition above?

Reply:
Wait, no-- the type param is passed through to `RetrievalService<T>`. So we can either leave it here, or stop extending `RetrievalService<T>` here (and in the mutable one) and have other classes implement `RetrievalService<T>` independently. WDYT?

Reply:
In that case, I'd probably leave this as-is for this PR. I wouldn't mind looking more closely at our use of those generics later -- they can probably be simplified, but that's really getting into a separate issue.

Reply:
Ok, cool. We can revisit, of course.